lib/ansible/modules/cloud/vmware/vmware_host_ntp.py

```diff
@@ -0,0 +1,265 @@
+        self.results = {}
+        self.desired_state = module.params['state']
+
+    def service_ctl(self, host, service_state):
+        changed = False
+        host_service_system = host.configManager.serviceSystem
+        if host_service_system:
+            try:
+                if service_state == 'start':
+                    host_service_system.StartService(id='ntpd')
+                if service_state == 'stop':
+                    host_service_system.StopService(id='ntpd')
+                if service_state == 'restart':
+                    host_service_system.RestartService(id='ntpd')
+                changed = True
+            except (vim.fault.InvalidState, vim.fault.NotFound, vim.fault.HostConfigFault) as e:
+                self.results[host.name]['error'] = e.msg
+
+        return changed
+
+    def update_ntp_settings(self, host, ntp_servers):
+        changed = False
+        host_date_time_manager = host.configManager.dateTimeSystem
+        if host_date_time_manager:
+            available_ntp_servers = host_date_time_manager.dateTimeInfo.ntpConfig.server
+            new_ntp_servers = available_ntp_servers + ntp_servers
+
+            ntp_config_spec = vim.host.NtpConfig()
+            ntp_config_spec.server = new_ntp_servers
+            date_config_spec = vim.host.DateTimeConfig()
+            date_config_spec.ntpConfig = ntp_config_spec
+            try:
+                host_date_time_manager.UpdateDateTimeConfig(date_config_spec)
+                changed = True
+            except vim.fault.HostConfigFault as e:
+                self.results[host.name]['error'] = e.msg
+            except Exception as e:
+                self.results[host.name]['error'] = e
+
+        return changed
+
+    def delete_ntp_settings(self, host, ntp_servers):
+        changed = False
+        host_date_time_manager = host.configManager.dateTimeSystem
+        if host_date_time_manager:
+            available_ntp_servers = host_date_time_manager.dateTimeInfo.ntpConfig.server
+            for server in ntp_servers:
+                if server in available_ntp_servers:
+                    available_ntp_servers.remove(server)
+
+            ntp_config_spec = vim.host.NtpConfig()
+            ntp_config_spec.server = available_ntp_servers
+            date_config_spec = vim.host.DateTimeConfig()
+            date_config_spec.ntpConfig = ntp_config_spec
+            try:
+                host_date_time_manager.UpdateDateTimeConfig(date_config_spec)
+                changed = True
+            except vim.fault.HostConfigFault as e:
+                self.results[host.name]['error'] = e.msg
+            except Exception as e:
+                self.results[host.name]['error'] = e
+
+        return changed
+
+    def check_host_state(self):
+        change_list = []
+        changed = False
+        for host in self.hosts:
+            state = 'absent'
+
+            host_ntp_servers = self.check_ntp_servers(host)
```

Comment:
@Akasurde Maybe you should change "host_ntp_servers" for "update_ntp_list" or "update_ntp_servers" or "ntp_servers_to_change" ("ntp_servers_to_add_or_remove").
What do you think ?

Reply:
OK.